editor/test/internal/node_test.clj

```diff
@@ -611,30 +606,551 @@
+                   (cacheable-regular-input-dependent-endpoints ov-consumer))
+                 (set (keys (g/cache))))))
+
+        (testing "Only array-input-producer-two connected to array-input."
+          (evaluate-and-check-array-input-dependents! consumer ["second change to array-input-producer-two"])
+          (evaluate-and-check-array-input-dependents! ov-consumer ["second change to array-input-producer-two"])
+          (is (= (set/union
+                   (cacheable-property-dependent-endpoints consumer)
+                   (cacheable-property-dependent-endpoints ov-consumer)
+                   (cacheable-regular-input-dependent-endpoints consumer)
+                   (cacheable-regular-input-dependent-endpoints ov-consumer)
+                   (cacheable-array-input-dependent-endpoints consumer)
+                   (cacheable-array-input-dependent-endpoints ov-consumer))
+                 (set (keys (g/cache))))))))))
+
+(g/defnode OverrideSuccessorsTestNode
+  (property property g/Any)
+  (input owning-input g/Any :cascade-delete :array)
+  (input external-input g/Any)
+
+  ;; Dependent on property.
+  (output output<=property g/Any (g/fnk [property] property))
+  (output secondary-output<=output<=property g/Any (g/fnk [output<=property] output<=property))
+
+  ;; Dependent on external-input.
+  (output output<=external-input g/Any (g/fnk [external-input] external-input))
+  (output secondary-output<=output<=external-input g/Any (g/fnk [output<=external-input] output<=external-input)))
+
+(defn- successor-output-endpoints
+  ([node-id output-label]
+   (successor-output-endpoints (g/now) node-id output-label))
+  ([basis node-id output-label]
+   (assert (g/has-output? (g/node-type* basis node-id) output-label) "Only outputs have successors.")
+   (let [graph-id (g/node-id->graph-id node-id)]
+     (into (sorted-set)
+           (get-in basis [:graphs graph-id :successors node-id output-label])))))
+
+(defn- dependent-internal-output-endpoints
+  ([node-id label]
+   (dependent-internal-output-endpoints (g/now) node-id label))
+  ([basis node-id label]
+   (let [node-type (g/node-type* basis node-id)
+         label->dependent-internal-output-labels (in/input-dependencies node-type)]
+     (into (sorted-set)
+           (map #(g/endpoint node-id %))
+           (label->dependent-internal-output-labels label)))))
+
+(defn- dependent-immediate-override-output-endpoints
+  ([node-id label]
+   (dependent-immediate-override-output-endpoints (g/now) node-id label))
+  ([basis node-id label]
+   (into (sorted-set)
+         (map #(gt/endpoint % label))
+         (g/overrides basis node-id))))
+
+(deftest override-successors
+  ;; This test recreates a scenario discovered when working on the gui system.
+  ;; We discovered an issue where the successors stored in the graph would not
+  ;; correct after adding a gui node to a scene imported as a template from
+  ;; another scene.
```

Comment:
This added test focuses on the successors in particular, and will fail if I remove the fix in `transaction.clj`.

Reply:
This is great!